src/MAT_HDF5.jl

```diff
@@ -336,15 +330,14 @@ function m_writearray{T<:HDF5BitsOrBool}(parent::HDF5Parent, name::String, adata
         close(dtype)
     end
 end
-function m_writearray{T<:HDF5BitsOrBool}(parent::HDF5Parent, name::String, adata::Array{Complex{T}})
+function m_writearray(parent::HDF5Parent, name::String, adata::Array{Complex{T}}) where {T<:HDF5BitsOrBool}
     dtype = build_datatype_complex(T)
     try
         stype = dataspace(adata)
         obj_id = HDF5.h5d_create(parent.id, name, dtype.id, stype.id)
         dset = HDF5Dataset(obj_id, file(parent))
         try
-            arr = reinterpret(T, adata, tuple(2, size(adata)...))
-            HDF5.writearray(dset, dtype.id, arr)
+            HDF5.writearray(dset, dtype.id, vec(adata))
```

Comment:
It might (I don't recall) but the more relevant point is that it shouldn't be necessary. Correct me if I'm wrong, but I don't think it matters to HDF5 what the originating type is, only what `dtype.id` says it is?